pinot-core/src/test/java/org/apache/pinot/core/util/CrcUtilsTest.java

```diff
@@ -38,8 +38,8 @@
 public class CrcUtilsTest {
   private static final File INDEX_DIR = new File(FileUtils.getTempDirectory(), "CrcUtilsTest");
   private static final String AVRO_DATA = "data/test_data-mv.avro";
-  private static final long EXPECTED_V1_CRC = 2708456273L;
-  private static final long EXPECTED_V3_CRC = 2796149869L;
+  private static final long EXPECTED_V1_CRC = 3087108662L;
```

Comment:
Why does this change?

Reply:
Since we need to add extra child field spec, so I refactor the logic and add method `addFieldSpec(properties, column, fieldSpec);`
However this changes the order of column stats in the `metadata.properties` files got changed.
Do you see this an issue for crc change?
E.g. 
before:
```
column.column1.cardinality = 51594
column.column1.totalDocs = 100000
column.column1.dataType = INT
column.column1.bitsPerElement = 16
column.column1.lengthOfEachEntry = 0
column.column1.columnType = DIMENSION
column.column1.isSorted = false
column.column1.hasDictionary = true
column.column1.isSingleValues = true
column.column1.maxNumberOfMultiValues = 0
column.column1.totalNumberOfEntries = 100000
column.column1.isAutoGenerated = false
column.column1.minValue = 15935
column.column1.maxValue = 2147475579
column.column1.defaultNullValue = -2147483648
```
after:
```
column.column1.cardinality = 51594
column.column1.totalDocs = 100000
column.column1.bitsPerElement = 16
column.column1.lengthOfEachEntry = 0
column.column1.isSorted = false
column.column1.hasDictionary = true
column.column1.maxNumberOfMultiValues = 0
column.column1.totalNumberOfEntries = 100000
column.column1.isAutoGenerated = false
column.column1.columnType = DIMENSION
column.column1.dataType = INT
column.column1.isSingleValues = true
column.column1.minValue = 15935
column.column1.maxValue = 2147475579
column.column1.defaultNullValue = -2147483648
```

Reply:
Ideally we want to keep it the same, or we might end up downloading segments because different versions of Pinot generating different crc

Reply:
Ok, reverted the code then, only use `addFieldSpec` for adding child FieldSpecs.